Projects/UOContent/Network/Packets/IncomingPlayerPackets.cs

```diff
@@ -420,27 +422,40 @@ public static void DisplayGumpResponse(NetState state, SpanReader reader)
                 return;
             }
 
-            var textEntries = new TextRelay[textCount];
+            scoped Span<ushort> textIds = default;
+            scoped Span<Range> textRanges = default;
+            ReadOnlySpan<byte> rawTextData = default;
 
-            for (var i = 0; i < textEntries.Length; ++i)
+            if (textCount > 0)
             {
-                int entryID = reader.ReadUInt16();
-                int textLength = reader.ReadUInt16();
+                int positionOffset = reader.Position;
 
-                if (textLength > 239)
+                textIds = stackalloc ushort[textCount];
+                textRanges = stackalloc Range[textCount];
+                rawTextData = reader.Buffer[positionOffset..];
+
+                for (var i = 0; i < textCount; ++i)
                 {
-                    state.LogInfo("Invalid gump response, disconnecting...");
-                    var exception = new InvalidGumpResponseException($"Text entry {i} is too long ({textLength})");
-                    exception.SetStackTrace(new StackTrace());
-                    NetState.TraceException(exception);
-                    state.Mobile?.SendMessage("Invalid gump response.");
-
-                    // state.Disconnect("Invalid gump response.");
-                    return;
-                }
+                    textIds[i] = reader.ReadUInt16();
+                    int textLength = reader.ReadUInt16();
 
-                var text = reader.ReadBigUniSafe(textLength);
-                textEntries[i] = new TextRelay(entryID, text);
+                    if (textLength > 239)
+                    {
+                        state.LogInfo("Invalid gump response, disconnecting...");
+                        var exception = new InvalidGumpResponseException($"Text entry {i} is too long ({textLength})");
+                        exception.SetStackTrace(new StackTrace());
+                        NetState.TraceException(exception);
+                        state.Mobile?.SendMessage("Invalid gump response.");
+
+                        // state.Disconnect("Invalid gump response.");
+                        return;
+                    }
+
+                    int textStart = reader.Position - positionOffset;
+                    textLength = reader.SkipString(TextEncoding.Unicode, textLength);
```

Comment:
We don't need a special method for this. We can just get the byte count per char from the encoding (it's always 2) and skip 2 × text length by using Seek from Current